src/App.tsx

```diff
@@ -15,48 +21,112 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+// eslint-disable-next-line max-len
+const stableSort = (array: string[], compareFunction: (a: string, b: string) => number) => {
+  return array
+    .map((item, index) => ({ item, index }))
+    .sort((a, b) => compareFunction(a.item, b.item) || a.index - b.index)
+    .map(({ item }) => item);
+};
```

Comment:
Your logic is good, but it can be simplified. Write one universal simple function that will return sorted data, based on sortType and isReversed. You need to use it to have sortedGoods in all places.